File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
import math

# 소수 판별
def is_prime(num):
    if num < 2: 
        return False
    for i in range(2, int(math.sqrt(num))+1):
        if num % i == 0:
            return False
    return True

def solution(numbers):
    new_numbers = [n for n in numbers] # 한 글자씩
    
    # 모든 경우의 수 만들기
    candidates = set()
    for i in range(1, len(numbers) + 1):
        for p in permutations(new_numbers, i):
            candidates.add(int(''.join(p)))
    
    return sum(1 for c in candidates if is_prime(c)) 

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>
#include <set>

using namespace std;

bool is_prime_number(int num) {
    if (num < 2) return false;
    for (int i=2; i < int(sqrt(num))+1; i++) {
        if (num % i == 0) return false;
    }
    
    return true;
}

int solution(string numbers) {
    sort(numbers.begin(), numbers.end()); // 순열을 돌리기 위한 정렬
    
    // 모든 경우의 수
    set<int> all_nums;
    do {
        string temp = "";
        for (int i=0; i < numbers.size(); i++) {
            temp += numbers[i];
            all_nums.insert(stoi(temp));
        }
        
    } while(next_permutation(numbers.begin(), numbers.end()));
    
    // 소수 개수 세기
    int answer = 0;
    for (auto num : all_nums) {
        if (is_prime_number(num)) answer ++;
    }
    
    return answer;
}